src/main/java/org/veupathdb/service/eda/ss/model/variable/VariableDataShape.java

```diff
@@ -20,4 +23,11 @@ public static VariableDataShape fromString(String shapeString) {
     }
     throw new RuntimeException("Unrecognized data shape: " + shapeString);
   }
+
+  public boolean isCompatibleWithCollectionShape(VariableDataShape collectionShape, Set<String> collectionVocab) {
+    if (this == BINARY || collectionShape == BINARY) {
```

Comment:
What about ordinal/categorical?  Wasn't that one of the mismatches?

Reply:
This should be caught in this case. If either of the variables are binary it just relies on the vocab size. All of the type-related cases in question involved binary/categorical variables.